input/drivers/dinput.c

```diff
@@ -181,6 +182,41 @@ static void *dinput_init(const char *joypad_driver)
    return di;
 }
 
+static void dinput_keyboard_mods(struct dinput_input *di, int mod)
+{
+   switch (mod)
+   {
+      case RETROKMOD_SHIFT:
+         if ( ((GetAsyncKeyState(VK_LSHIFT) >> 1) && !di->shift_l) ||
```

Comment:
Is it possible you can cache the results of GetAsyncKeyState(VK_LSHIFT) here once instead of having to call it twice ? Of course, if you do this, make sure that the code is still C89-compliant by declaring the temporary variable either at the start of a code block or the start of the function.